datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedTaskCountEstimator.java

```diff
@@ -65,8 +65,11 @@ public int getTaskCount(ClusterThroughputInfo throughputInfo, List<String> assig
         LoadBasedPartitionAssignmentStrategyConfig.DEFAULT_PARTITION_MESSAGES_IN_RATE, "");
     // total throughput in KB/sec
     int totalThroughput = allPartitions.stream()
-        .map(p -> throughputMap.getOrDefault(p, defaultThroughputInfo))
-        .mapToInt(PartitionThroughputInfo::getBytesInKBRate)
+        .mapToInt(p ->  {
+          String wildcardTopicPartition = p.substring(0, p.lastIndexOf('-')) + "-" + "-1";
```

Comment:
Appending `--1` to the topic name seems a bit arbitrary. If the intent is to save a topic level throughput info in the same map, how about just using the topic name (without appended `--1`)?

Reply:
Thank you for the suggestion. I did not think from this angle. It will work. I have refactored the code. Thanks again.